planning/planners/src/encode.rs

```diff
@@ -630,11 +647,23 @@ pub fn encode(pb: &FiniteProblem, metric: Option<Metric>) -> anyhow::Result<(Mod
                 ConstraintType::Or => {
                     let mut disjuncts = Vec::with_capacity(constraint.variables.len());
                     for v in &constraint.variables {
-                        let disjunct: Lit = Lit::try_from(*v)?;
+                        let disjunct: Lit = Lit::try_from(*v).context("or")?;
                         disjuncts.push(disjunct);
                     }
                     model.bind(or(disjuncts), value)
                 }
+                ConstraintType::Sum(sum) => {
+                    //println!("debug linear: {:?} = {}", lsum, value);
+
+                    //let value = sum.value;
+                    let sum = sum.sum.clone();
+
+                    //model.enforce(sum.clone().leq(value), []);
+                    //model.enforce(sum.geq(value), []);
+
+                    model.enforce(sum.clone().leq(LinearSum::zero()), []);
```

Comment:
The `value` is not used.

Reply:
I agree. It would simplifies the struct used, however it could also be non-trivial for another programmer, as it would force him to rearrange the sum to have all terms on the right side of the sum.

Reply:
Well this is a mostly internal API. Specialized API for this should be done on the LinearSum object instead.
E.g. an `eq` method that transforms it into a `LinearEq`

Reply:
After rebasing you should be able to access an improved API for linear sums.